File: Data Structures & Algorithms/duplicate-integer/submission-0.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        unordered_map<int, int> ans;
        for(int x : nums){
            ans[x]++;
        }
        for(auto[key,val] : ans){
            if (val>1 ){
                return true;
            }
        }
        return false;
    }
};

File: Data Structures & Algorithms/combination-target-sum/submission-0.py
class Solution:
    def combinationSum(self, nums: List[int], target: int) -> List[List[int]]:
        ans = []
        curr = 0
        curarr = []
        i = 0
        def dfs(currlist, i, total):
            if total > target or i >= len(nums):
                return 
            if total == target:
                ans.append(currlist.copy())
                return
            currlist.append(nums[i])
            dfs(currlist, i, total+nums[i])
            currlist.pop()
            dfs(currlist, i+1, total)

        dfs([], 0, 0)
        return ans
            

